src/submission/NodeSubmissionClient.ts

```diff
@@ -16,44 +16,35 @@ export class NodeSubmissionClient extends DefaultSubmissionClient {
     this.configurationVersionHeader = this.configurationVersionHeader.toLowerCase();
   }
 
-  public sendRequest(config:Configuration, method:string, path:string, data:string, callback: (status:number, message:string, data?:string, headers?:Object) => void): void {
-    function complete(response:http.IncomingMessage, responseBody:string, responseHeaders:Object) {
-      var message:string;
-      if (response.statusCode === 0) {
-        message = 'Unable to connect to server.';
-      } else if (response.statusCode < 200 || response.statusCode > 299) {
-        message = response.statusMessage || (<any>response).message;
-      }
+  public sendRequest(config: Configuration, method: string, path: string, data: string, callback: NodeSubmissionCallback): void {
+    var request: NodeSubmissionRequest = {
+      method,
+      path,
+      data,
+      serverUrl: config.serverUrl,
+      apiKey: config.apiKey,
+      userAgent: config.userAgent
+    };
 
-      callback(response.statusCode || 500, message, responseBody, responseHeaders);
+    var exitController = config.exitController;
+    if (exitController.isApplicationExiting) {
+      this.sendRequestSync(request, callback);
+    } else {
+      submitRequest(request, callback);
     }
+  }
 
-    var parsedHost = url.parse(config.serverUrl);
-    var options:https.RequestOptions = {
-      auth: `client:${config.apiKey}`,
-      headers: {},
-      hostname: parsedHost.hostname,
-      method: method,
-      port: parsedHost.port && parseInt(parsedHost.port),
-      path: path
-    };
+  private sendRequestSync(request: NodeSubmissionRequest, callback: NodeSubmissionCallback): void {
```

Comment:
I am not an expert with Node. Guessing this is the only way to do a synchronous http request?